### PR TITLE
2015 day 11 part 1

### PR DESCRIPTION
**Pull request by Herrera93**
Created a password generator library

### CODE REVIEW

**Comment by Herrera93** [comment]
[2015-day-11-part-1](https://trello.com/c/kCRkNi2s/14-2015-day-11-part-1)

File: tests/day_11.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+extern crate advent_of_code;
+use advent_of_code::year2015::day_11::{find_password, is_valid_password};
+
+#[test]
+fn test_day_11_one() {
```
Comment (kanekv) [inline tests/day_11.rs]: let's have 1-line tests in a single test section. too many tests just bloat up the code

File: src/year2015/day_11.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+    password_ints.iter().map(|c| *c as char).collect::<String>()
+}
+
+fn skip_bad_values(collection: &mut [u8]) {
+    let mut found = false;
+    for val in collection {
+        if found {
+            *val = 97;
```
Comment (kanekv) [inline src/year2015/day_11.rs]: add comments what those "magical" numbers mean

File: src/year2015/day_11.rs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        } else {
+            counter = 1;
+        }
+        previous_value = *val;
+    }
+    counter == 3
+}
+
+fn contains_bad_values(collection: &[u8]) -> bool {
+    collection.iter().any(|&x| x == 105 || x == 108 || x == 111)
```
Comment (kanekv) [inline src/year2015/day_11.rs]: add comment here too

File: src/year2015/day_11.rs (hunk 2 above)
Comment (Herrera93) [reply src/year2015/day_11.rs]: Added constants for the "magic" numbers and a couple of comments for clarification.

**Pull request merged**